src/main/java/de/tum/in/www1/artemis/service/connectors/BitbucketService.java

```diff
@@ -91,11 +91,47 @@ public void configureRepository(URL repositoryUrl, String username) {
         }
 
         giveWritePermission(getProjectKeyFromUrl(repositoryUrl), getRepositorySlugFromUrl(repositoryUrl), username);
-        protectMasterBranch(getProjectKeyFromUrl(repositoryUrl), getRepositorySlugFromUrl(repositoryUrl), username);
+        protectBranches(getProjectKeyFromUrl(repositoryUrl), getRepositorySlugFromUrl(repositoryUrl));
     }
 
-    private void protectMasterBranch(String projectKeyFromUrl, String repositorySlugFromUrl, String username) {
-        // TODO: Simon Leiß implement, prevent deletion and prevent force push
+    private void protectBranches(String projectKey, String repositorySlug) {
```

Comment:
Maybe we can add some JavaDoc to this method, explaining that it will set up two restrictions on all branches in that repository?

Reply:
@jpbernius I've added some documentation